src/main/java/jarvis/logic/Parser.java

```diff
@@ -0,0 +1,109 @@
+                String toFind = input.substring(5);
+                tasklist.find(toFind);
+            } //else {
+//                try {
+//                    tasklist.add(input);
+//                    String result = tasklist.acknowledge();
+//                } catch (ArrayIndexOutOfBoundsException e) {
+//                    System.out.println("bad input: invalid format " + input);
+//                } catch (IndexOutOfBoundsException e) {
+//                    System.out.println("bad input: invalid format " + input);
+//                } catch (IllegalArgumentException e) {
+//                } catch (Exception e) {
+//                    System.out.println("error");
+//                }
+//            }
+        }
+    }
+
+    /**
+     * Parses the input command and returns the corresponding result as a dialog string.
+     *
+     * @param input    the user input string.
+     * @param tasklist the TaskList to operate on.
+     * @return the dialog string corresponding to the input command.
+     */
+    public static String parseDialog(String input, TaskList tasklist) {
+        StringBuilder result = new StringBuilder();
+
+        if (input.equals("list")) {
+            result.append(tasklist.list());
+        } else if (!input.equals("bye")) {
+            // Only return if it's not "bye"
+            if (input.startsWith("mark") || input.startsWith("unmark") ||
+                    input.startsWith("delete")) {
+                String[] parts = input.split(" ");
+                try {
+                    if (parts.length == 2) {
+                        int taskIndex = Integer.parseInt(parts[1]);
+                        if (input.startsWith("mark")) {
+                            result.append(tasklist.markDialog(taskIndex));
+                        } else if (input.startsWith("unmark")) {
+                            result.append(tasklist.unmarkDialog(taskIndex));
+                        } else if (input.startsWith("delete")) {
+                            result.append(tasklist.deleteDialog(taskIndex));
+                        }
+                    }
+                } catch (IndexOutOfBoundsException e) {
+                    result.append("index error!\n");
+                }
+
+            } else if (input.startsWith("find")) {
+                String toFind = input.substring(5).trim();
+                result.append(tasklist.find(toFind));
+            } else {
+                try {
+                    result.append(tasklist.add(input)).append("\n");
+                } catch (ArrayIndexOutOfBoundsException e) {
+                    result.append("bad input: invalid format ").append(input).append("\n");
+                } catch (IndexOutOfBoundsException e) {
+                    result.append("bad input: invalid format ").append(input).append("\n");
+                } catch (IllegalArgumentException e) {
+                    result.append("Invalid input.\n");
+                } catch (Exception e) {
```

Comment:
I'm not too sure if I'm correct on this, but I think Pokemon exception handling here wouldn't be ideal as a catch-all statement because we wouldn't know where went wrong if there is a new exception expect. What do you think?